src/gg.regression.unity.bots/Runtime/Scripts/RGTestUtils.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Collections;
+using RegressionGames.StateRecorder;
+using UnityEngine;
+using UnityEngine.Assertions;
+using UnityEngine.SceneManagement;
+using Object = UnityEngine.Object;
+
+namespace RegressionGames
+{
+    /// <summary>
+    /// Utilities for running tests within the Unity Test Runner using Regression Games features.
+    /// </summary>
+    public class RGTestUtils
+    {
+        
+        /// <summary>
+        /// Waits until a specific scene has been loaded, and asserts that it has been loaded
+        /// </summary>
+        /// <param name="sceneName">The name of the scene to wait for</param>
+        /// <param name="timeout">The maximum time to wait for the scene to load (in seconds). Defaults to 5.</param>
+        public static IEnumerator WaitForScene(string sceneName, int timeout = 5)
+        {
+            var startTime = Time.time;
+            bool loaded = false;
+            while (!loaded)
+            {
+                var scene = SceneManager.GetSceneByName(sceneName);
+                loaded = scene.IsValid() && scene.isLoaded;
+                if (!loaded)
+                {
+                    var timeElapsed = Time.time - startTime;
+                    if (timeElapsed > timeout)
+                    {
+                        break;
+                    }
+                }
+                yield return null;
+            }
+            yield return null;
+            Assert.IsTrue(loaded);
+        }
+        
+        
+        /// <summary>
+        /// Plays back an existing recording, and then returns the save location of the recording.
+        /// </summary>
+        /// <param name="recordingPath">The path to the recording to play back (the full data.zip file path)</param>
+        /// <param name="setSaveLocation">A callback that will be called with the save location of the recording</param>
+        public static IEnumerator StartPlaybackFromFile(string recordingPath, Action<string> setSaveLocation)
```

Comment:
I think using a callback here is a reasonable approach to returning values. I do not believe there is a clean way to directly return values from coroutines in the way you were intending 

A suggestion: You may want to replace the `Action<string>` callback with a more general `Action<PlaybackEvent>` callback, since in the future you may want to notify the testing script with more types of events besides just "playback finished". For example, you may want to allow the test script to be notified whenever a new keyframe is encountered.